Covid Track/Covid Track/ViewModels/Visit History/VisitViewModel.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  VisitViewModel.swift
+//  Covid Track
+//
+//  Created by Samip shah on 1/8/20.
+//  Copyright © 2020 Samip shah. All rights reserved.
+//
+
+import UIKit
+
+class VisitViewModel {
+  private let visitModel: VisitModel
+  
+  var profile: UIImage? {
+    if let data = visitModel.visitor?.profile{
+      return UIImage(data: data)
+    }
+    return nil
+  }
+  
+  var name: String? {
+    visitModel.visitor?.name
+  }
+  
+  var email: String? {
+    visitModel.visitor?.email
+  }
+  
+  var phone: String? {
+    visitModel.visitor?.phone
+  }
+  
+  var home: String? {
+    visitModel.visitor?.homeAddress
+  }
+  
+  var work: String? {
+    visitModel.visitor?.workAddress
+  }
+  
+  var temperatureLevel: String {
+    print(visitModel.temperatureLevel)
+    let tempLevel = TemperatureLevel(rawValue:  visitModel.temperatureLevel )
+    switch tempLevel {
+    case .normal:
+      return "Normal"
+      
+    case .high:
+      return "High"
+      
+    case .danger:
+      return "Danger"
+      
+    default:
+      return "NA"
+    }
+  }
+  
+  var temperatureDescription: String? {
+    let temperature = TemperatureLevel(rawValue: visitModel.temperatureLevel)
+    return temperature?.description
+  }
+  
+  var temperatureColor: UIColor {
+    let tempLevel = TemperatureLevel(rawValue: visitModel.temperatureLevel )
+    return tempLevel?.color ?? .black
+  }
+  
+  var checkInTime: String? {
+    if let date = visitModel.checkIn {
+      return readableTimeFrom(date: date)
+    }
+    return nil
+  }
+  
+  var checkInTimeDescription: String? {
+    if let date = visitModel.checkIn{
+      return getTimeDescription(date: date)
+    }
+    return nil
+  }
+  
+  var checkOutTime: String? {
+    if let date = visitModel.checkOut {
+      return readableTimeFrom(date: date)
+    }
+    return nil
+  }
+  
+  init(_ visitModel: VisitModel) {
+    self.visitModel = visitModel
+  }
+  
+  private func readableTimeFrom(date: Date) -> String{
+    let timeInterval = Date().timeIntervalSince(date)
+    var description = ""
+   
+    let second = 1.0
+    let min = 60 * second
+    let hour = 60 * min
+    let day = 24 * hour
+    let year = 365 * day
```

Comment:
Use proper setter and getter as per required

Reply:
Those are the constants used for calculating time ago, and their scope is inside a function only, so I believe it doesn't need to be changed.